src/collectors/proc.plugin/sys_class_power_supply.c

```diff
@@ -116,14 +121,82 @@ static void add_labels_to_power_supply(struct power_supply *ps, RRDSET *st) {
     rrdlabels_add(st->rrdlabels, "device", ps->name, RRDLABEL_SRC_AUTO);
 }
 
+static void read_simple_property(struct power_supply *ps, struct simple_property *prop, bool keep_fds_open) {
+    char buffer[30 + 1];
+
+    if(unlikely(prop->fd == -1)) {
+        prop->fd = open(prop->filename, O_RDONLY | O_CLOEXEC, 0666);
+        if(unlikely(prop->fd == -1)) {
+            collector_error("Cannot open file '%s'", prop->filename);
+            power_supply_free(ps);
+            ps = NULL;
+        }
+    }
+
+    if (ps)
+    {
+        ssize_t r = read(prop->fd, buffer, 30);
+        if(unlikely(r < 1)) {
+            collector_error("Cannot read file '%s'", prop->filename);
+            power_supply_free(ps);
+            ps = NULL;
+        }
+        else {
+            buffer[r] = '\0';
+            prop->value = str2ull(buffer, NULL);
+
+            if(unlikely(!keep_fds_open)) {
+                close(prop->fd);
+                prop->fd = -1;
+            }
+            else if(unlikely(lseek(prop->fd, 0, SEEK_SET) == -1)) {
+                collector_error("Cannot seek in file '%s'", prop->filename);
+                close(prop->fd);
+                prop->fd = -1;
+            }
+        }
+    }
+}
+
+static void rrdset_create_simple_prop(struct power_supply *ps, struct simple_property *prop, char *title, char *dim, collected_number divisor, char *units, long priority, int update_every) {
+    if(unlikely(!prop->st)) {
+        char id[RRD_ID_LENGTH_MAX + 1], context[RRD_ID_LENGTH_MAX + 1];
+        snprintfz(id, RRD_ID_LENGTH_MAX, "powersupply_%s", dim);
+        snprintfz(context, RRD_ID_LENGTH_MAX, "powersupply.%s", dim);
+
+        prop->st = rrdset_create_localhost(
+                id
+                , ps->name
+                , NULL
+                , dim,
```

Comment:
```suggestion
                , dim
```

I see this error in ci logs

```c /netdata/src/collectors/proc.plugin/sys_class_power_supply.c: In function 'rrdset_create_simple_prop':
/netdata/src/collectors/proc.plugin/sys_class_power_supply.c:180:9: error: macro "rrdset_create_localhost" passed 13 arguments, but takes just 12
```

Reply:
Thanks. Not sure how I missed it. :-)